src/components/content/allergies/helpers/allergies-filter.tsx

```diff
@@ -1,21 +1,30 @@
 import { AllergyModel } from "@/fhir/models/allergies";
 import { ResourceMap } from "@/fhir/types";
-import { isEqual, uniqWith } from "@/utils/nodash";
 import { sort } from "@/utils/sort";
 
 export const applyAllergyFilters = (
   data: fhir4.AllergyIntolerance[],
+  builderId: string,
   includedResources?: ResourceMap
 ) => {
   const allergyModel = data.map((allergy) => new AllergyModel(allergy, includedResources));
 
   const sortedByDate = sort(allergyModel, "recordedDate", "desc", true);
 
-  const allergyData = uniqWith(sortedByDate, (a, b) =>
-    isEqual(valuesToDedupeOn(a), valuesToDedupeOn(b))
-  );
+  // pull out unique allergies preferring those owned by the builder
+  const allergyDataMap = new Map<string, AllergyModel>();
+  sortedByDate.forEach((allergy) => {
+    const { lowercaseDisplay } = allergy;
+    const existingAllergy = allergyDataMap.get(lowercaseDisplay);
 
-  return allergyData;
-};
+    if (!existingAllergy) {
+      allergyDataMap.set(lowercaseDisplay, allergy);
+    } else if (allergy.ownedByBuilder(builderId) && !existingAllergy.ownedByBuilder(builderId)) {
+      allergyDataMap.set(lowercaseDisplay, allergy);
+    }
```

Comment:
Regardless of which condition matches, they both do the same `allergyDataMap.set(lowercaseDisplay, allergy)`. Perhaps it's broken out into multiple if statements for readability, if so we could use variables for the conditions to summarize
```suggestion
    const allergyOwnedByBuilder = allergy.ownedByBuilder(builderId);
    const existingAllergyNotOwned = !existingAllergy.ownedByBuilder(builderId);
    if (!existingAllergy || (allergyOwnedByBuilder && existingAllergyNotOwned)) {
      allergyDataMap.set(lowercaseDisplay, allergy);
    }
```